packages/alfa-rules/src/common/predicate/has-computed-style.ts

```diff
@@ -0,0 +1,28 @@
+import { Device } from "@siteimprove/alfa-device";
+import { Element, Text } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Context } from "@siteimprove/alfa-selector";
+import { Property, Style } from "@siteimprove/alfa-style";
+
+const { isElement } = Element;
+
+export function hasComputedStyle<N extends Property.Name>(
+  device: Device,
+  context?: Context
+): (
+  name: N,
+  predicate: Predicate<Style.Computed<N>>
+) => Predicate<Element | Text> {
```

Comment:
```suggestion
export function hasComputedStyle<N extends Property.Name>(
  name: N, 
  predicate: Predicate<Style.Computed<N>>,
  device: Device,
  context?: Context
): Predicate<Element | Text> {
```

Is there a compelling reason to optimise for a partially applied function? If not, let's not wrap things in an additional closure.

Reply:
Yep, wasn't sure about that one… My first version was uncurryfied (https://github.com/Siteimprove/alfa/pull/848/commits/0b65f47453948011a9489a2cf5f124536fd27a73) but then I thought the call site would be much better if we can partially apply the common stuff (https://github.com/Siteimprove/alfa/pull/848/files#diff-cddeee40baf90b48e519a73729f0571a0544d5e7b80fa7fabf6350e2b8702f49R51-R54) but then again, this does result in an ugly partial application due to eta-expansion of type variables (https://github.com/Siteimprove/alfa/pull/848/files#diff-cddeee40baf90b48e519a73729f0571a0544d5e7b80fa7fabf6350e2b8702f49R33-R36)

So I wasn't sure about it and I guess I'll just uncurryfy it again 😁